Core.DomainServices/Queries/Contract/QueryReadModelsWhichShouldExpire.cs

```diff
@@ -21,13 +22,7 @@ public IQueryable<ItContractOverviewReadModel> Apply(IQueryable<ItContractOvervi
 
                 x =>
                     // All currently set as active in the read model
-                    x.IsActive &&
-                    // Remove results where the date has no effect (active overrides all other logic)
-                    x.SourceEntity.Active == false &&
-                    // Expiration data defined
-                    x.SourceEntity.ExpirationDate != null &&
-                    // Expiration date has passed
-                    x.SourceEntity.ExpirationDate < currentTime || (x.SourceEntity.Terminated != null && x.SourceEntity.Terminated < currentTime)
+                    x.IsActive && ItContractIsActiveQueryHelper.CheckIfContractIsExpired(currentTime, x.SourceEntity)
```

Comment:
no no no... you are breaking the expression here... it's an expression tree which must be compilable into SQL.. that is killed by this now.

Reply:
If you are going to create a helper you need a lot more type contraints and the helper should generate an expression tree